consumer/consumererror/throttle.go

```diff
@@ -0,0 +1,53 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package consumererror // import "go.opentelemetry.io/collector/consumer/consumererror"
+
+import (
+	"errors"
+	"time"
+)
+
+// TODO: Clean this by forcing all exporters to return an internal error type that always include the information about retries.
+
+// throttled wraps an error to indicate that it should be retried after a delay
+type throttled struct {
+	err   error
+	delay time.Duration
+}
+
+func (t throttled) Error() string {
+	return "Throttle (" + t.delay.String() + "), error: " + t.err.Error()
+}
+
+func (t throttled) Unwrap() error {
+	return t.err
+}
+
+// ThrottleRetryDelay returns the duration of time we should wait to retry, or
+// zero if the error is not a throttled retry error
+func ThrottleRetryDelay(err error) time.Duration {
```

Comment:
Why this instead of exposing "Throttle" type and add a func on it `RetryDelay`?

Reply:
Moved to #5569